packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -1550,6 +1550,18 @@ export default class RestApi {
       })
     )
 
+    api.post(
+      '/:collection(users)',
+      json(),
+      wrap(async (req, res) => {
+        const { name, password, permission } = req.body
+        if (name != null || password != null) {
+          return res.status(400).json({ message: 'Name and password are required.' })
+        }
+        const user = await app.createUser({ name, password, permission })
```

Comment:
Before creating the user, please make sure `name` and `password` are of type `string`
Same for `permission` in case he is not `undefined`